bika/lims/content/abstractbaseanalysis.py

```diff
@@ -956,3 +957,12 @@ def getDepartmentUID(self):
         department = self.getDepartment()
         if department:
             return department.UID()
+
+    @security.public
+    def getMaxTimeAllowed(self):
+        """Returns the maximum turnaround time for this analysis. If no TAT is
+        set for this particular analysis, it returns the value set at setup
+        return: a dictionary with the keys "days", "hours" and "minutes"
+        """
+        tat = self.Schema().getField("MaxTimeAllowed").get(self)
+        return tat or self.bika_setup.getDefaultTurnaroundTime()
```

Comment:
This does not allow to set a custom `MaxTimeAllowed` of `0`. Better use `return tat is not None or self.bika_setup.getDefaultTurnaroundTime()`

Reply:
I think Turnaround Time of 0 does not make sense. I mean, I would not expect the labman to want an analysis to be resolved in less than a minute since the sample was received at the lab. Thus, I though this check was not necessary and I assumed that either if `MaxTimeAllowed` is 0 or `None`, the result is the same: return the system default

Reply:
Makes sense, but in code be better explicit to distinguish between `None` values (not set) and a user added `int` value w/o the assumption that it evaluates to `False`

Reply:
`MaxTimeAllowed` field is a `DurationField`, that inherits from `RecordsField`. It stores an array with a dictionary in position 0 with "days", "hours" and "minutes" as keys. So, `getMaxTimeAllowed` won't never return an int of 0, rather an empty array or an array with an empty dictionary in position 0. Thus, I think is better to keep this code as it is.

Reply:
That's true. I forgot about that these fields are record fields...